client/mailing-lists/utils.js

```diff
@@ -1,48 +1,29 @@
-/**
- * External dependencies
- */
-
-import debugFactory from 'debug';
-
-const debug = debugFactory( 'calypso:mailing-lists' );
-
 /**
  * Internal dependencies
  */
 import wpcom from 'calypso/lib/wp';
 
-export default {
-	deleteSubscriber( category, emailAddress, hmac, context ) {
-		return new Promise( function ( resolve, reject ) {
-			wpcom
-				.undocumented()
-				.mailingList( category )
-				.unsubscribe( emailAddress, hmac, context, function ( err, result ) {
-					if ( err ) {
-						debug( err );
-						reject( err );
-						return;
-					}
-
-					resolve( result );
-				} );
-		} );
-	},
+function createSubscriberResourceUrl( category, emailAddress, method ) {
+	return (
+		'/mailing-lists/' +
+		encodeURIComponent( category ) +
+		'/subscribers/' +
+		encodeURIComponent( emailAddress ) +
+		'/' +
+		method
```

Comment:
It's fine that we don't support `method` to be optional anymore - all usages of `createSubscriberResourceUrl` are always providing it.